src/python/pants/engine/env_vars.py

```diff
@@ -17,6 +17,9 @@
 class CompleteEnvironmentVars(FrozenDict):
     """CompleteEnvironmentVars contains all environment variables from the current Pants process.
 
+    For non-local environments like docker or remote execution we also include the local environment
+    from the shell running pants.
```

Comment:
I think we should be careful with this. When executing in a remote environment, there is no guarantee that my env will be compatible with the remote system. 

Think we should require the user to be very explicit with which env vars they want forwarded to the remote system.

Reply:
+1 Do you think this could look like an extra field on the `remote_environment` and/or `docker_environment` target?

Something like
`pass_through_env_vars=["ENV_VAR_1", "ENV_VAR_2"]`
and would mean that those variables are safe to pass through from the local environment and made available in the non-local environment

Reply:
something like that, yea. feels like a good bikeshed topic for slack, being a user facing thing.. ;)

Reply:
Yea, definitely a per-environment config field.

Reply:
@stuhood @kaos , made the change. It is now controlled via a new field `pass_through_env_vars`